VK_V1.0/VK_V1.0/Scenes/MyGroups/MyGroupsTableViewController.swift

```diff
@@ -57,22 +57,18 @@ final class MyGroupsTableViewController: UITableViewController {
     // MARK: - Private methods
 
     private func loadMyGroups() {
-        createNotificationToken()
-        networkService.fetchMyGroups { [weak self] results in
-            guard let self = self else { return }
-            switch results {
-            case let .success(result):
-                DataProvider.save(items: result.response.items)
-                self.groups = self.dataProvider.loadData(items: Group.self)
-                self.tableView.reloadData()
-            case .failure:
-                self.showErrorAlert(title: Constants.errorTitleString, message: Constants.errorDescriptionString)
-            }
-        }
+        networkService.getGroups()
+        fetchGroups()
+    }
+
+    private func fetchGroups() {
+        guard let group = dataProvider.loadData(items: Group.self) else { return }
```

Comment:
```suggestion
        guard let group = dataProvider.loadData(items: Group.self) else { return }
```
не понятно, я думала там что то, отвечающее за распределение или фильтрацию данных, а там рилм со своими методами...  я б переименовала бы

Reply:
если хочешь получать данные от дата провайдера, сделай так, чтобы контроллер о Риме тоже не знал. чтобы он вообще не понимал, с ритма или с Нетворк они приходят